### PR TITLE
Enable Short to Integer converters

### PR DESCRIPTION
**Pull request by Elishtar**
I haven't forgotten about:
- [x] Tests
- [x] Changelog
- [ ] Documentation
- [x] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:

### CODE REVIEW

File: src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java [hunk 1]
```diff
@@ -76,8 +79,7 @@ private DefaultMessagePackMapperFactory() {
                         new DefaultExtensionValueToBigDecimalConverter())
                 .withValueConverter(ValueType.NIL, Object.class, new DefaultNilValueToNullConverter())
                 //TODO: add this when will it be resolved https://github.com/tarantool/cartridge-java/issues/118
```
Comment (ArtDu) [inline src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java:81]: Delete it

File: src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java [hunk 2]
```diff
@@ -76,8 +79,7 @@ private DefaultMessagePackMapperFactory() {
...
-                // .withValueConverter(ValueType.INTEGER, Short.class, new DefaultIntegerValueToShortConverter())
-                // .withObjectConverter(Short.class, new DefaultShortToIntegerObjectConverter())
+                .withObjectConverter(Short.class, IntegerValue.class, new DefaultShortToIntegerValueConverter())
```
Comment (ArtDu) [inline src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java:82]: Move it to numeric types converters. It's more semantically correct

File: src/test/java/io/tarantool/driver/mappers/DefaultIntegerValueToShortConverterTest.java [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import io.tarantool.driver.mappers.converters.value.DefaultIntegerValueToShortConverter;
+import org.junit.jupiter.api.Test;
+import org.msgpack.value.ValueFactory;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+/**
+ * @author Andrey Klyuev
+ */
+public class DefaultIntegerValueToShortConverterTest {
```
Comment (ArtDu) [inline src/test/java/io/tarantool/driver/mappers/DefaultIntegerValueToShortConverterTest.java:12]: The test's already existed. See DefaultShortConverterTest

**Review by ArtDu — CHANGES_REQUESTED** [review]

File: src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java (hunk 2 above)
Comment (Elishtar) [reply src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java]: Something like this?

File: src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java (hunk 1 above)
Comment (Elishtar) [reply src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java]: Renamed, I think we should have this TODO, for info if something went wrong

File: src/test/java/io/tarantool/driver/mappers/DefaultIntegerValueToShortConverterTest.java (hunk 3 above)
Comment (Elishtar) [reply src/test/java/io/tarantool/driver/mappers/DefaultIntegerValueToShortConverterTest.java]: Removed

File: src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java (hunk 2 above)
Comment (ArtDu) [reply src/main/java/io/tarantool/driver/mappers/DefaultMessagePackMapperFactory.java]: Yeap, it's all right

File: src/test/java/io/tarantool/driver/mappers/DefaultIntegerValueToShortConverterTest.java (hunk 3 above)
Comment (Elishtar) [reply src/test/java/io/tarantool/driver/mappers/DefaultIntegerValueToShortConverterTest.java]: Of course

**Review by ArtDu — APPROVED** [review]

**Pull request merged**